packages/plugin-svelte/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import path from 'path';
+import { logger } from '@rsbuild/shared';
+import type { RsbuildPlugin, RsbuildPluginAPI } from '@rsbuild/core';
+
+export type PluginSvelteOptions = {};
+
+export function pluginSvelte(
+  options: PluginSvelteOptions = {},
+): RsbuildPlugin<RsbuildPluginAPI> {
+  return {
+    name: 'plugin-svelte',
+
+    async setup(api) {
+      let sveltePath: string = '';
+      try {
+        // Resolve `svelte` package path from the project directory
+        sveltePath = path.dirname(
+          require.resolve('svelte/package.json', {
+            paths: [api.context.rootPath],
+          }),
+        );
+      } catch (err) {
+        logger.error(
+          `Cannot resolve \`svelte\` package under the project directory, did you forget to install it?`,
+        );
+        throw new Error(`Cannot resolve \`svelte\` package`, {
+          cause: err,
+        });
+      }
+
+      api.modifyRsbuildConfig((config, { mergeRsbuildConfig }) => {
+        return mergeRsbuildConfig(config, {
+          source: {
+            alias: {
+              svelte: path.join(sveltePath, 'src/runtime'),
+            },
+          },
+        });
+      });
+
+      api.modifyBundlerChain(async (chain, { CHAIN_ID, isProd }) => {
+        chain.resolve.extensions.add('.svelte');
+
+        const loaderPath = require.resolve('svelte-loader');
+
+        // We need to set an alias from `svelte-loader` to its realpath.
+        // Because with `emitCss` option on, the loader generates css
+        // imports with inline loader reference like `!svelte-loader...`,
+        // which would cause the bundler failed to resolve the loader.
+        // See https://github.com/sveltejs/svelte-loader/blob/344f00744b06a98ff5ee7e7a04d5e04ac496988c/index.js#L128
+        chain.merge({
+          resolveLoader: {
+            alias: {
+              'svelte-loader': loaderPath,
+            },
+          },
+        });
+
+        chain.module
+          .rule(CHAIN_ID.RULE.SVELTE)
+          .test(/\.svelte$/)
+          .use(CHAIN_ID.USE.SVELTE)
+          .loader(loaderPath)
+          .options({
+            compilerOptions: {
+              dev: !isProd,
+            },
+            emitCss: isProd,
```

Comment:
Can we use output.disableCssExtract to control the `emitCss` option?

https://rsbuild.dev/config/options/output.html#outputdisablecssextract